spock-gradle/src/main/groovy/org/spockframework/gradle/SpockBasePlugin.groovy

```diff
@@ -28,7 +27,9 @@ class SpockBasePlugin implements Plugin<Project> {
       def configFile = project.file("Spock${taskName}Config.groovy")
       if (configFile.exists()) {
         task.inputs.file(configFile)
-        task.systemProperty "spock.configuration", configFile
+          .withPropertyName("spockConfig")
+          .withPathSensitivity(PathSensitivity.RELATIVE)
+        task.systemProperty "spock.configuration", project.relativePath(configFile)
```

Comment:
This tracks the OS-dependent relative path as input via `systemProperty`. It would be better to use a `CommandLineArgumentProvider` here.